File: Analizador Semantico/Analizador/src/Aritmetico.java
import TokenType;
import Token;
import Nodo;
import SymbolTable as ts;
import sys;

public class SolverAritmetico {
    public SolverAritmetico() {

    }

    public Object resolver(Nodo n) {
        if (n.hijos == null) {
            if (n.value.type == TokenType.NUMBER || n.value.type == TokenType.STRING) {
                return n.value.literal;
            } else if (n.value.type == TokenType.IDENTIFIER) {
                return ts.simbolos.obtener(n.value.lexeme);
            } else {
                return null;
            }
        }
        Nodo nizq = n.hijos[0];
        Nodo nder = n.hijos[1];
        Object rizquierdo = resolver(nizq);
        Object rderecho = resolver(nder);
        if (rizquierdo instanceof Float && rderecho instanceof Float) {
            switch (n.value.type) {
                case TokenType.ADD:
                    return (Float)rizquierdo + (Float)rderecho;
                case TokenType.SUB:
                    return (Float)rizquierdo - (Float)rderecho;
                case TokenType.MULT:
                    return (Float)rizquierdo * (Float)rderecho;
                case TokenType.DIAG:
                    return (Float)rizquierdo / (Float)rderecho;
            }
        } else if (rizquierdo instanceof String && rderecho instanceof String) {
            if (n.value.type == TokenType.ADD) {
                return (String)rizquierdo + (String)rderecho;
            }
        } else {
            System.out.println("Error: No se puede resolver la operacion " + n.value.type + " con las instancias " + rizquierdo.getClass().getName() + " y " + rderecho.getClass().getName());
        }
        return null;
    }
}

File: AnalizadorSem/Aritmetico.py
from Tokentype import TokenType
from Token import Token
from Nodo import Nodo
import SymbolTable as ts
import sys


class SolverAritmetico:
    def __init__(self) -> None:
        pass

    def resolver(self, n: Nodo):
        if n.hijos is None:
            if n.value.type == TokenType.NUMBER or n.value.type == TokenType.STRING:
                return n.value.literal

            elif n.value.type == TokenType.IDENTIFIER:
                return ts.simbolos.obtener(n.value.lexeme)
            else:
                return None

        nizq: Nodo = n.hijos[0]
        nder: Nodo = n.hijos[1]

        rizquierdo = self.resolver(nizq)
        rderecho = self.resolver(nder)

        if isinstance(rizquierdo, float) and isinstance(rderecho, float):
            match n.value.type:
                case TokenType.ADD:
                    return rizquierdo + rderecho
                case TokenType.SUB:
                    return rizquierdo - rderecho
                case TokenType.MULT:
                    return rizquierdo * rderecho
                case TokenType.DIAG:
                    return rizquierdo / rderecho
        elif isinstance(rizquierdo, str) and isinstance(rderecho, str):
            if n.value.type == TokenType.ADD:
                return rizquierdo + rderecho
        else:
            print(F"Error: No se puede resolver la operacion {n.value.type} con las instancias {type(rizquierdo)} y {type(rderecho)}")
        return None